include/cpp-statsd-client/StatsdClient.hpp

```diff
@@ -59,28 +60,57 @@ class StatsdClient {
     const std::string& errorMessage() const noexcept;
 
     //! Increments the key, at a given frequency rate
-    void increment(const std::string& key, float frequency = 1.0f) const noexcept;
+    void increment(const std::string& key,
+                   float frequency = 1.0f,
+                   const std::vector<std::string>& tags = {}) const noexcept;
```

Comment:
`std::vector` is fine but do you know if there is something preventing us from using a variadic argument instead?
Could that be easier to use somehow?

Reply:
the main reason is that `frequency` has a default. so the compiler says, "why don't your var args have a default too"

Reply:
the other option is to use variadic templates but that is much messier than looping over a vector i think. should i give that a try though just in case, ive not really used them before so it might be fun to try it out!

Reply:
No need to dig too much into it. As I said, `std::vector` is fine, so let's go with that :+1: